internal/dkg/state_machine.go

```diff
@@ -679,7 +682,9 @@ func isValidStateChange(current, next Status) bool {
 	case TimedOut:
 		return next == Proposing || next == Proposed || next == Aborted
 	case Failed:
-		return next == Proposing || next == Proposed || next == Joined || next == Left || next == Aborted
+		// a node can be `Failed` but still be included in the group file under some (magical) circumstances.
+		// In such a case, it should be added as a remainer on the next DKG rather than a joiner.
```

Comment:
I kinda disagree: if a node failed to join, they don't have a valid share and so cannot act as a proper "remainer" in the DKG. 
Are we sure such a node is going to use the "new joiner" codepath despite them being "remainer" here?

Reply:
an example could be if they have a network timeout; their node will be in a failed state, though the other nodes will seem in the group file. 
Given they're in the group file, if another user proposes a new DKG, all the other nodes will expect them to be a `Remainer`. Unfortunately "alive for the DKG" is completely separate from "send a valid partial at any point after the DKG".
The failed node won't need to use the new joiner path as they'll still have the node list from previous steps